pkg/rpc/response/result/invoke.go

```diff
@@ -76,6 +108,29 @@ func (r *Invoke) UnmarshalJSON(data []byte) error {
 			if err != nil {
 				break
 			}
+			if st[i].Type() == stackitem.InteropT {
+				iteratorAux := new(iteratorAux)
+				if json.Unmarshal(arr[i], iteratorAux) == nil {
+					iteratorValues := make([]stackitem.Item, len(iteratorAux.Value))
+					for j := range iteratorValues {
+						iteratorValues[j], err = stackitem.FromJSONWithTypes(iteratorAux.Value[j])
+						if err != nil {
+							err = fmt.Errorf("failed to unmarshal iterator values: %w", err)
+							break
+						}
+					}
+					// it's impossible to restore initial iterator type; also iterator is almost
+					// useless outside of the VM, thus let's replace it with the most general
+					// iterable type (Array)
+					// TODO: @roman-khimov, option #2 is to create iterator over array:
```

Comment:
We need to decide which one would be better to use. At the current moment, option#1 is implemented.

Reply:
Option 2 won't work after #1935. And we need to keep `truncated` flag somehow. So maybe we need to store some special `struct` there with `truncated` and `values`? Anyone could try the type of the `Value()` then and see if it's the thing he expects.

Reply:
Nice solution, implemented.